api/search.go

```
@@ -1,28 +1,644 @@
+	TextPrev           string    `json:"textPrev"` // the previous subtitle line
+	Text               string    `json:"text"`
+	TextNext           string    `json:"textNext"` // the next subtitle line
+	StreamName         string    `json:"streamName"`
+	StreamStartTime    time.Time `json:"streamStartTime"`
+	StreamEndTime      time.Time `json:"streamEndTime"`
+	CourseName         string    `json:"courseName"`
+	CourseSlug         string    `json:"courseSlug"`
+	CourseYear         int       `json:"year"`
+	CourseTeachingTerm string    `json:"semester"`
+}
+
+// checkAndFillResponse takes the response of meilisearch and filters out all courses/streams/subtitles which the user is not allowed to see (checking)
+// this is necessary because updating the course in the database (e.g. changing the visibility for a course from public to loggedin) does not update meilisearch data until the next day
+// additionally it adds information to the results which is not saved in meili (fill)
+// ---
+// canSearchHiddenCourses indicates whether the response may include streams or subtitles of a hidden course
+// should only be true when the user has explicitly named the hidden course he wants to search through in the url params
+func checkAndFillResponse(c *gin.Context, user *model.User, limit int64, daoWrapper dao.DaoWrapper, response *meilisearch.MultiSearchResponse, canSearchHiddenCourses bool) {
+	var userEligibleToSeeResultsOfHiddenCourse func(course model.Course) bool
+	if canSearchHiddenCourses {
+		userEligibleToSeeResultsOfHiddenCourse = user.IsEligibleToWatchCourse
+	} else {
+		userEligibleToSeeResultsOfHiddenCourse = user.IsEligibleToSearchForCourse
+	}
+
+	for i, res := range response.Results {
+		switch res.IndexUID {
+		case "STREAMS":
+			hits := res.Hits
+			res.Hits = []any{}
+			response.Results[i] = meilisearch.SearchResponse{}
+
+			var meiliStreams []SearchStreamDTO
+			temp, err := json.Marshal(hits)
+			if err != nil { // shouldn't happen
```

There is an awfull lot of ignoring errors here and in other parts.
Logging these errors might be helpfull..